### ISSUE TITLE
Invalid archive produced when using "Archive" in Xcode 9 (with keyboard extension)

### ISSUE THREAD

**Issue by taylorwrynn** [CocoaPods/CocoaPods#7301]
CocoaPods 1.3.1
  Xcode 9
 

It likes #4021. Removing "copy pod resources" from extension target is OK.

The Podfile:

```ruby
target app do
  use_frameworks!
  pod 'Masonry', '~>1.1.0'
  pod 'ReactiveObjC', '~> 3.0.0'
end

target appex do
  use_frameworks!
  pod 'ReactiveObjC', '~> 3.0.0'
  pod 'Masonry', '~>1.1.0'
end
```

**Comment by LiamInJapan** [comment]
I'm having the same problem (though in my case I have no keyboard extension)

18:19:22 ✔ liam:(master)~/Development/ANSO-Townville/iOS$ pod --version
1.3.1

**Comment by dnkoutso** [comment]
You can use a `post_install` hook to edit the `Pods.xcodeproj` if you want to override a setting.

**Comment by bigyelow** [comment]
https://developer.apple.com/library/content/technotes/tn2215/_index.html#//apple_ref/doc/uid/DTS40011221-CH1-PROJ

Check the following reasons:

- Your archive contains header files.
- Your archive contains static libraries or frameworks.

**Comment by mixable** [comment]
Same for me. In my app, there are two extensions (an Action Extension and a Today Extension).

I use **CocoaPods 1.5.0** and **Xcode Version 9.3 (9E145)**.

I tried and checked different things, but without success:
- Removing "Copy Pod Resource" build steps in extensions
- Setting `SKIP_INSTALL` to "Yes" in all Targets / in all Pods and "No" in Projects
- Product clean and removing all DerivedData before Archive
- 'Your archive contains header files' as mentioned in [Troubleshooting Application Archiving in Xcode](https://developer.apple.com/library/content/technotes/tn2215/_index.html#//apple_ref/doc/uid/DTS40011221-CH1-PROJ) > there is no "Headers" section listed in Build Phases
- ...

The major change I made since my last release was to switch some dynamic libraries/frameworks to static. Everything compiles and runs without errors. Just the Archiving is not possible anymore.

@bigyelow what else is necessary to handle static libraries or frameworks correctly?
